src/Symfony/Component/Serializer/Normalizer/NumberNormalizer.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use BcMath\Number;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+
+/**
+ * Normalizes a {@see Number} to a string.
+ */
+final class NumberNormalizer implements NormalizerInterface, DenormalizerInterface
+{
+    public function getSupportedTypes(?string $format): array
+    {
+        return [
+            Number::class => true,
+        ];
+    }
+
+    public function normalize(mixed $data, ?string $format = null, array $context = []): string
+    {
+        if (!$data instanceof Number) {
+            throw new InvalidArgumentException(\sprintf('The data must be an instance of "%s".', Number::class));
+        }
+
+        return (string) $data;
+    }
+
+    public function supportsNormalization(mixed $data, ?string $format = null, array $context = []): bool
+    {
+        return $data instanceof Number;
+    }
+
+    /**
+     * @throws NotNormalizableValueException
+     */
+    public function denormalize(mixed $data, string $type, ?string $format = null, array $context = []): Number
+    {
+        if (Number::class !== $type) {
+            throw new InvalidArgumentException(\sprintf('Only "%s" type is supported.', Number::class));
+        }
+
+        if (!\is_int($data) && !\is_string($data)) {
```

Comment:
```suggestion
        if (!\is_int($data) && !\is_string($data)) {
```

Nit: I would test for strings first as it will likely be the most popular type (maybe a `match`/`switch` could also be more elegant).

Reply:
It is a valid point, it's changed now. I also changed the order of types in the error message to reflect that `string` is usually the expected type though `int` works too.

Tried to convert this to `match` also, but I found the separate `if`'s easier to read after all since the error messages are so long.